File: test2.js
// const UserClosure = ({ firstName, lastName, age, occupation }) => {
//   return {
//     describeSelf: () => {
//       console.log(
//         `My name is ${firstName} ${lastName}, I am ${age}years Old and i work as a ${occupation}`
//       );
//     },
//     getAge: () => {
//       return age;
//     },
//     showStrength: () => {
//       let howOld = age;
//       let output = "I am";
//       while (howOld-- > 0) {
//         output += " very";
//       }
//       return output + " Strong";
//     },
//   };
// };

const showStrength = age => {
  let howOld = age;
  let output = "I am";
  while (howOld-- > 0) {
    output += " very";
  }
  return output + " Strong";
};

const userClosure = ({ firstName, lastName, age, occupation }) => {
  return {
    describeSelf: () => {
      console.log(
        `My name is ${firstName} ${lastName}, I am ${age}years Old and i work as a ${occupation}`
      );
    },
    getAge: () => {
      return age;
    },
    showStrength: () => showStrength(age),
  };
};
const individuals = [];
for (let i = 0; i < 10000; i++) {
  const person = userClosure({
    firstName: "Zaynab",
    lastName: "Olagunju",
    age: [i],
    occupation: "Economist",
  });
  individuals.push(person);
}
const used = process.memoryUsage();
for (let key in used) {
  console.log(`${key} ${Math.round((used[key] / 1024 / 1024) * 100) / 100} MB`);
}
const start = Date.now();
individuals.map(person => person.showStrength());
console.log(
  "Finished displaying strength in " + (Date.now() - start) / 1000 + " seconds"
);
// rss 30.12 MB heapTotal 18.23 MB heapUsed 8.03 MB external 0.01 MB
// Finished displaying strength in 4.037 seconds

File: test2.py
from timeit import default_timer as timer


def show_strength(age):
    how_old = age
    output = "I am "
    while how_old > 0:
        output += "very "
        how_old -= 1

    return output + "strong"


def describe_self(first_name, last_name):
    print(f"name{first_name} {last_name}")


start_time = timer()

individuals = []
for i in range(10000):
    person = {
        "first_name": "Zay",
        "last_name": "olag",
        "age": i,
        "occupation": "Economist",
    }
    individuals.append(person)


for p in individuals:
    # print(show_strength(p["age"]))
    show_strength(p["age"])


end_time = timer()

execution_time = end_time - start_time
print(f"The execution time is: {execution_time}")
